### PR TITLE
feat: add accessibility state to focusable views

### PR DESCRIPTION
**Pull request by JulienIzz**
This PR adds an example of a test in the example app. Its purpose is to show how to test component of a real app.

### CODE REVIEW

File: packages/example/src/modules/program/view/ProgramList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    </ThemeProvider>,
+  );
+};
+
+describe('ProgramList', () => {
+  it('renders the list of item and scrolls accordingly when doing RIGHT twice', async () => {
+    const component = renderWithProviders(<ProgramList />);
+
+    const listElement = await component.findByTestId('program-list');
+    expect(listElement).toHaveStyle({
```
Comment (pierpo) [inline packages/example/src/modules/program/view/ProgramList.test.tsx:28]: - you should check that the element that you'd expect is focused
- and you should query it by its accessibility label ;)

Comment (JulienIzz) [reply packages/example/src/modules/program/view/ProgramList.test.tsx]: Done :D

File: packages/lib/src/spatial-navigation/components/FocusableView.tsx [hunk 2]
```diff
@@ -82,7 +82,10 @@ type InnerFocusableViewProps = FocusableViewProps & {
 
 const InnerFocusableView = forwardRef<View, InnerFocusableViewProps>(
   ({ viewProps, webProps, children, isActive, isFocused, style }, ref) => {
-    const accessibilityProps = useSpatialNavigatorFocusableAccessibilityProps();
+    const accessibilityProps = {
+      ...useSpatialNavigatorFocusableAccessibilityProps(),
+      accessibilityState: { selected: isFocused },
+    };
```
Comment (pierpo) [inline packages/lib/src/spatial-navigation/components/FocusableView.tsx:88]: don't do it like this, it won't be memoized ;)

in that regard, we should maybe include it in useSpatialNavigatorFocusableAccessibilityProps (that does memoize it) but that might break things, safest would be to leave it here in a separate memo

```tsx
const accessibilityState = useMemo(() => ({ selected: isFocused }), [isFocused])
```

Comment (JulienIzz) [reply packages/lib/src/spatial-navigation/components/FocusableView.tsx]: Done :)

**Pull request merged**